jax/api.py

```diff
@@ -857,11 +865,25 @@ def pmap(fun, axis_name=None, devices=None, backend=None):
   """
   _check_callable(fun)
   axis_name = _TempAxisName(fun) if axis_name is None else axis_name
+  if isinstance(static_argnums, int):
+    static_argnums = (static_argnums,)
 
   @wraps(fun)
   def f_pmapped(*args, **kwargs):
     f = lu.wrap_init(fun)
-    args, in_tree = tree_flatten((args, kwargs))
+    if static_argnums:
+      # Check no static arguments have been sharded
+      for i in static_argnums:
+        if type(args[i]) is pxla.ShardedDeviceArray:
```

Comment:
I think this type-based check is incorrect: a ShardedDeviceArray shouldn't be treated any differently from a regular DeviceArray or ndarray, since logically they represent the same thing.